### PR TITLE
Teacher login

### PR DESCRIPTION
**Pull request by ekat-br**
This is my PR for [US12](https://github.com/ekat-br/roomcheck/issues/26). Thanks in advance for reviewing my code. 

I'm so sorry for changing over 40 files. I had a boring weekend and started refactoring my code - that's why there are so many files. The part for the teacher login has 4 changed files - you see them them in the first 4 commits.

### CODE REVIEW

File: components/Button/styles.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import styled from "styled-components";
 
-const StyledEditButton = styled.button`
+const StyledButton = styled.button`
   display: inline-block;
```
Comment (CJAkkuec) [inline components/Button/styles.js:4]: Quick little note: there are [cases](https://css-tricks.com/when-do-you-use-inline-block/#aa-buttons) for a button to be an inline-block, but I'm not entirely sure if your use cases would fit the bill here. I'm not really asking you to change it, I'm just curious. 👀

File: components/Button/styles.js [hunk 2]
```diff
@@ -10,17 +10,18 @@ const StyledEditButton = styled.button`
   width: 200px;
   line-height: 45px;
   border-radius: 7px;
+  margin-top: 15px;
```
Comment (CJAkkuec) [inline components/Button/styles.js:13]: I'd probably advise against built-in margins in your components, unless absolutely required. If you need space around your buttons, there's probably a different way to handle that. (Maybe this is where the inline-block display mode gets tricky.)

File: components/Login/index.js [hunk 3]
```diff
@@ -4,31 +4,39 @@ import {
...
-  AdminButton,
+  RoleButton,
 } from "./styles";
 import Router from "next/router";
-import Link from "next/link";
 
 export default function Login() {
   const router = Router;
   const { data: session } = useSession();
+  console.log("Session", session);
```
Comment (CJAkkuec) [inline components/Login/index.js]:
```suggestion
```
Please make sure to yeet `console.log`s and commented out code before pushing a commit. :)

File: components/Navigation/styles.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ const NavContainer = styled.nav`
   right: 0;
```
Comment (CJAkkuec) [inline components/Navigation/styles.js:13]:
```suggestion
```
I'm not sure if this does anything? 🤔

**Review by CJAkkuec — CHANGES_REQUESTED** [review]
Sidenote: Refactoring is handled as its own issue and shouldn't be part of a feature PR. You wouldn't write a user story for it, but just a regular ticket, which only includes your developer tasks (the one you usually list at the bottom of a user story).

Looks good to me overall, with a couple of notes. There's probably some compositional and architectural stuff that could be handled a smidge differently (such as button variants instead of multiple button components), but that may be part of the polish phase when you're all done. 👍

File: components/Button/styles.js (hunk 1 above)
Comment (ekat-br) [reply components/Button/styles.js:4]: I take this for the next user story which focuses only on refactoring the buttons. Thank you for the feedback!

File: components/Button/styles.js (hunk 2 above)
Comment (ekat-br) [reply components/Button/styles.js:13]: Thank you, I will try to find a better way. This is part of the next user story.

**Review by moonwave99 — APPROVED** [review]
an easy LGTM!

**Pull request merged**